vault/resource_kubernetes_secret_backend_role_test.go

```diff
@@ -0,0 +1,167 @@
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, consts.FieldBackend, backend),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldName, name),
+					resource.TestCheckResourceAttr(resourceName, fieldAllowedKubernetesNamespaces+".#", "2"),
+					resource.TestCheckResourceAttr(resourceName, fieldAllowedKubernetesNamespaces+".0", "dev"),
+					resource.TestCheckResourceAttr(resourceName, fieldAllowedKubernetesNamespaces+".1", "int"),
+					resource.TestCheckResourceAttr(resourceName, fieldKubernetesRoleType, "Role"),
+					resource.TestCheckResourceAttr(resourceName, fieldGeneratedRoleRules, "rules:\n- apiGroups: [\"\"]\n  resources: [\"pods\"]\n  verbs: [\"list\"]\n"),
+					resource.TestCheckResourceAttr(resourceName, fieldServiceAccountName, ""),
+					resource.TestCheckResourceAttr(resourceName, fieldTokenMaxTTL, "12h"),
+					resource.TestCheckResourceAttr(resourceName, fieldTokenDefaultTTL, "6h"),
+					resource.TestCheckResourceAttr(resourceName, fieldExtraLabels+".%", "2"),
+					resource.TestCheckResourceAttr(resourceName, fieldExtraLabels+".id", "abc123"),
+					resource.TestCheckResourceAttr(resourceName, fieldExtraLabels+".name", "some_name"),
+					resource.TestCheckResourceAttr(resourceName, fieldExtraAnnotations+".%", "2"),
+					resource.TestCheckResourceAttr(resourceName, fieldExtraAnnotations+".env", "development"),
+					resource.TestCheckResourceAttr(resourceName, fieldExtraAnnotations+".location", "earth"),
+				),
+			},
+			{
+				Config: testKubernetesSecretBackendRole_UpdateConfig2(backend, name),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, consts.FieldBackend, backend),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldName, name),
+					resource.TestCheckResourceAttr(resourceName, fieldAllowedKubernetesNamespaces+".#", "1"),
+					resource.TestCheckResourceAttr(resourceName, fieldAllowedKubernetesNamespaces+".0", "*"),
+					resource.TestCheckResourceAttr(resourceName, fieldGeneratedRoleRules, ""),
+					resource.TestCheckResourceAttr(resourceName, fieldServiceAccountName, ""),
+					resource.TestCheckResourceAttr(resourceName, fieldKubernetesRoleType, "Role"),
+					resource.TestCheckResourceAttr(resourceName, fieldKubernetesRoleName, "existing_role"),
+					resource.TestCheckResourceAttr(resourceName, fieldTokenMaxTTL, "0s"),
+					resource.TestCheckResourceAttr(resourceName, fieldTokenDefaultTTL, "0s"),
+					resource.TestCheckResourceAttr(resourceName, fieldExtraLabels+".%", "0"),
+					resource.TestCheckResourceAttr(resourceName, fieldExtraAnnotations+".%", "0"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{fieldServiceAccountJWT},
+			},
+		},
+	})
+}
+
+func testAccKubernetesSecretBackendRoleCheckDestroy(s *terraform.State) error {
```

Comment:
Suggest replacing with https://github.com/hashicorp/terraform-provider-vault/blob/c520061319a277600ea367adacdcf7e17d2820ee/vault/testhelpers_test.go#L14

Reply:
Oops, yeah you are right, I think this comment was meant for the backend resource.